lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -335,3 +338,47 @@ def get(self, request, program_uuid):
                 'credit_pathways': credit_pathways,
             }
         )
+
+
+class CourseRecommendationApiView(APIView):
+
+    """
+        **Example Request**
+
+        GET api/dashboard/v0/recommendation/courses/
+    """
+
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """
+            Retrieves course recommendations details of a user in a specified course.
+        """
+        user_id = request.user.id
+        is_control, course_keys = get_personalized_course_recommendations(user_id)
+
+        # Emits an event to track student dashboard page visits.
+        segment.track(
+            user_id,
+            'edx.bi.student-dashboard-page.viewed',
```

Comment:
@shafqatfarhan Can we rename this to something more specific to recommendations instead dashboard viewed? This name doesn't make sense inside an Amplitude recommendations get API view.

Reply:
@waheedahmed may be `edx.bi.recommendations.viewed` will be suitable here?